### PR TITLE
Package overhaul

### PR DESCRIPTION
**Pull request by nabobalis**
Fixes https://github.com/LM-SAL/aiapy/issues/315

TODO:

- [x] Changelog for the min python change to 3.9
- [x] Mention removal of setup and need of modern pip for this to work.
- [x] Mention the use of * in some functions now
- [x] Fix CI

### CODE REVIEW

File: licenses/SUNPY.rst
Comment (wtbarnes) [inline licenses/SUNPY.rst:1]: Why was this deleted? I included this originally because the code in the `register` function was copied directly from `sunpy`.

File: pyproject.toml [hunk 1]
```diff
@@ -1,33 +1,128 @@
...
-            "wheel"]
+[project]
+name = "aiapy"
+dynamic = ["version"]
+description = "Python library for AIA data analysis."
+readme = "README.rst"
+requires-python = ">=3.9"
+license = {file = "LICENSE.txt"}
+keywords = ["solar physics", "solar", "science", "NASA", "SDO", "AIA", "solar dynamics observatory",
+            "atmospheric imaging assembly"]
```
Comment (wtbarnes) [inline pyproject.toml:17]: Break these up into one line each?

**Review by wtbarnes — APPROVED** [review]
Looks good to me. Thanks for taking this on. Just a few clarification questions.

**Comment by wtbarnes** [comment]
I noticed that RTD is still not triggering on this PR.

File: aiapy/__init__.py [hunk 2]
```diff
@@ -1,8 +1,8 @@
 from pathlib import Path
-from .version import version as __version__
-
 from itertools import compress
 
+from .version import version as __version__
```
[the remark this answers is not in the document]
Comment (nabobalis) [reply aiapy/__init__.py:4]: It was always there I think, just moved.

File: CODE_OF_CONDUCT.md [hunk 3]
```diff
@@ -48,15 +48,71 @@ comments, commits, code, wiki edits, issues, and other contributions that are
...
 Examples of representing our community include using an official e-mail address,
 posting via an official social media account, or acting as an appointed
 representative at an online or offline event.
 
-## Attribution
+## Enforcement ##
+
+Instances of abusive, harassing, or otherwise unacceptable behavior may be
+reported to the community leaders responsible for enforcement at
+[INSERT CONTACT METHOD].
```
[the remark this answers is not in the document]
Comment (nabobalis) [reply CODE_OF_CONDUCT.md:63]: Good point, I decided to update the COC to the latest version but maybe I should just link to the sunpy one?

**Comment by nabobalis** [comment]
> I noticed that RTD is not reporting its build status back to this PR. Is that just because there hasn't been a push since it was configured for this repo?

closed and reopened, both it and codecov are now live

File: pyproject.toml (hunk 1 above)
Comment (nabobalis) [reply pyproject.toml:17]: LIKE BLACK?!

**Comment by nabobalis** [comment]
That RTD build is just spinning: Here is the older one https://aiapy--313.org.readthedocs.build/en/313/ with the same content.

**Review by wtbarnes — COMMENTED** [review]
I'm a little concerned about the additional docs material, specifically naming it a "hands-on guide". The docs may need a refactor, but this should be left to another PR.

File: docs/tutorial/index.rst [hunk 4]
```diff
@@ -0,0 +1,16 @@
+.. _aiapy-handson-index:
+
+********************
+aiapy hands on guide
```
[the remark this answers is not in the document]
Comment (nabobalis) [reply docs/tutorial/index.rst:4]: Feels too brief for a tutorial.

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+    uses: OpenAstronomy/github-actions-workflows/.github/workflows/tox.yml@main
+    with:
+      default_python: '3.9'
+      submodules: false
+      coverage: codecov
+      toxdeps: tox-pypi-filter
+      posargs: -n auto --dist loadgroup
+      libraries: |
+        apt:
+          - libopenjp2-7
```
[the remark this answers is not in the document]
Comment (nabobalis) [reply .github/workflows/ci.yml:79]: Woops.

File: docs/tutorial/index.rst (hunk 4 above)
[the remark this answers is not in the document]
Comment (nabobalis) [reply docs/tutorial/index.rst:4]: You want me to pull this all out into a new PR?

**Comment by nabobalis** [comment]
Doc changes removed.
